grep-0015-remove-swig.md

```diff
@@ -27,18 +29,156 @@ anyway.
 ## Copyright / License
 
 This GREP is licensed as CC-BY-ND.
-Copyright 2018 The GNU Radio Foundation.
+Copyright 2020 Free Software Foundation??
 
 ## Motivation
 
 A lot of us have cursed SWIG programming. If we can get rid of it... well,
 wouldn't that be great. But maybe we can't. Let's find out.
 
+Since SWIG takes in the entire public header file, many methods and class members that are not necessarily intended to be exposed through the python API gets automatically bound.  Pybind11 requires deliberate binding of methods, and the binding code does not get regenerated on every compile.  
+
 ## Description
 
-**To be written**. GREP14 (blocktool) would be a help.
+### Block Binding Code Structure
+
+In each gr module, create a `python/bindings` directory
+
+`python/bindings`
+-  `python_bindings.cpp` <-- c++ code to build up the pybind11 module block by block
+   -  alternatively this can be broken up into a per-block cpp file
+-  `generated/` <-- folder to contain automatically generated binding code with [blockname]_python.hpp
+-  `custom/` <-- folder to contain manually generated binding code
+
+Let's take as an example the public header of char_to_float.h in gr-blocks/include/gnuradio/blocks
+
+```c++
+class BLOCKS_API char_to_float : virtual public sync_block
+{
+public:
+    typedef std::shared_ptr<char_to_float> sptr;
+    static sptr make(size_t vlen = 1, float scale = 1.0);
+    virtual float scale() const = 0;
+    virtual void set_scale(float scale) = 0;
+};
+```
+
+In order to bind this, we write/generate `char_to_float_python.hpp`:
```

Comment:
Let's just make this char_to_float_python.cc, and treat as a separate compilation unit